consensus/src/simplex/actors/voter/actor.rs

```diff
@@ -341,13 +341,31 @@ impl<E: Clock, S: Scheme, D: Digest> Round<E, S, D> {
         Some(finalization)
     }
 
-    /// Returns whether at least one honest participant has notarized a proposal.
-    pub fn at_least_one_honest(&self) -> Option<View> {
-        if self.notarizes.len() <= self.scheme.participants().max_faults() as usize {
-            return None;
+    /// Returns true if the proposal in this view has a valid ancestry (to determine if we should backfill
+    /// missing certificates implied by a proposal).
+    ///
+    /// We determine this by checking if at least one honest participant notarized the proposal for
+    /// this view. That is, if any of the following are true:
+    /// - a finalization for this view exists, or
+    /// - a notarization certificate for this view exists, or
+    /// - the number of notarize votes exceeds the maximum number of faulty participants.
+    pub fn proposal_has_valid_ancestry(&self) -> bool {
```

Comment:
This function name now implies something about `proposal`'s verification status, which i don't like.